core/shared/platform/common/posix/posix_socket.c

```diff
@@ -166,13 +201,28 @@ os_socket_shutdown(bh_socket_t socket)
 }
 
 int
-os_socket_inet_network(const char *cp, uint32 *out)
+os_socket_inet_network(uint8 ipv4_or_ipv6, const char *cp, uint8 *out)
```

Comment:
Same as above, how about `bool is_ipv4`?
And for `uint8 *out`, there may be address unaligned issue, if the input is not 4-byte aligned, L218, `*ipv4_buffer = ntohl(*ipv4_buffer);` may cause core dump in some targets, e.g. arm?

How about define an union type, like:
```C
union {
    uint32 ipv4;
    uint16 ipv6[8];
};
```

Reply:
Good point, updated.